### ISSUE TITLE
MoonGen freezes a machine with 82598EB 10G card

### ISSUE THREAD

**Issue by guerby** [emmericp/MoonGen#106]
Hi,

When running Moogen from git on a core i7 2600 (BIOS HT disabled) 16G RAM ubuntu 14.04 kernel 3.13 and network card :

01:00.0 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)
01:00.1 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)

01:00.0 0200: 8086:10f1 (rev 01)
	Subsystem: 8086:a21f

root@t1:~/MoonGen# ./build/MoonGen ./examples/tx-multi-core.lua 
EAL: Detected lcore 0 as core 0 on socket 0
EAL: Detected lcore 1 as core 1 on socket 0
EAL: Detected lcore 2 as core 2 on socket 0
EAL: Detected lcore 3 as core 3 on socket 0
EAL: Support maximum 64 logical core(s) by configuration.
EAL: Detected 4 lcore(s)
EAL: Setting up memory...
EAL: Ask a virtual area of 0xc00000 bytes
EAL: Virtual area found at 0x7f669d600000 (size = 0xc00000)
EAL: Ask a virtual area of 0x800000 bytes
EAL: Virtual area found at 0x7f669cc00000 (size = 0x800000)
EAL: Ask a virtual area of 0x200000 bytes
EAL: Virtual area found at 0x7f669c800000 (size = 0x200000)
EAL: Ask a virtual area of 0x200000 bytes
EAL: Virtual area found at 0x7f669c400000 (size = 0x200000)
EAL: Ask a virtual area of 0x1b400000 bytes
EAL: Virtual area found at 0x7f6677200000 (size = 0x1b400000)
EAL: Ask a virtual area of 0x63000000 bytes
EAL: Virtual area found at 0x7f6614000000 (size = 0x63000000)
EAL: Ask a virtual area of 0x200000 bytes
EAL: Virtual area found at 0x7f669c000000 (size = 0x200000)
EAL: Ask a virtual area of 0x200000 bytes
EAL: Virtual area found at 0x7f669bc00000 (size = 0x200000)
EAL: Requesting 1024 pages of size 2MB from socket 0
EAL: TSC frequency is ~3411484 KHz
EAL: Master core 0 is ready (tid=a19fa8c0)
EAL: Core 3 is ready (tid=9b3fe700)
EAL: Core 2 is ready (tid=9bbff700)
EAL: Core 1 is ready (tid=9eb45700)
EAL: PCI device 0000:01:00.0 on NUMA socket -1
EAL:   probe driver: 8086:10f1 rte_ixgbe_pmd
EAL:   0000:01:00.0 not managed by VFIO driver, skipping
EAL:   PCI memory mapped at 0x7f66a19a3000
EAL:   PCI memory mapped at 0x7f66a1963000
EAL:   PCI memory mapped at 0x7f66a195f000

Here machine freezes completely I have to power off and on.

On the same machine DPDK 2.2.0 (from tarball) and pktgen-dpdk (from git) work fine.

**Comment by emmericp** [comment]
can you try the dpdk2.0 branch?

**Comment by guerby** [comment]
I did :

```
root@t3:~# mkdir new
root@t3:~# cd new
root@t3:~/new# git clone -b dpdk2.0 https://github.com/emmericp/MoonGen.git
root@t3:~/new# cd MoonGen/
root@t3:~/new/MoonGen# git submodule update --init
root@t3:~/new/MoonGen# ./build.sh
root@t3:~/new/MoonGen# ./bind-interfaces.sh 
root@t3:~/new/MoonGen# ./build/MoonGen ./examples/tx-multi-core.lua 
```

And after a few seconds the machine froze, no lines printed after the last command.

**Comment by emmericp** [comment]
That's the same bug, the dpdk2.0 branch just doesn't show the output.

Not sure what is happening here. I hope to find some time to port everything to dpdk 2.2 which should fix that for you.

**Comment by guerby** [comment]
I will test soon on a machine with a X520 controller 82599EN, I'll let you know.

What is the currently recommanded version of DPDK for MoonGen from git?

**Comment by guerby** [comment]
Moogen from git main branch works fine with the X520 so this is a 82598EB specific issue.

```
root@t1:~/MoonGen# ./build/MoonGen ./examples/tx-multi-core.lua 1 10000 1 
...
[Queue 0] Sent 1249953201 packets, current rate 14.88 Mpps, 7618.77 MBit/s, 9999.63 MBit/s wire rate
```

**Comment by guerby** [comment]
Reopen

**Comment by emmericp** [comment]
Oh, I misread your initial post. A 82598 is quite rare and I think I know what might ne causing this. But I'll have to check the datasheet...

Never got my hands on a 82598, I only have a lot of 82599s (aka X520)

**Comment by guerby** [comment]
If you wish to debug this issue I can give you ssh access to our not-for-profit lab with 82598EB, just mail me with your ssh key attached (not inline) at

laurent AT guerby DOT net

Otherwise feel free to close this issue.

**Comment by emmericp** [comment]
I think this might be fixed in 41dd4d0
